src/pnt.rs

```diff
@@ -0,0 +1,199 @@
+use crate::dpq::r_dt_0;
+use crate::dpq::r_dt_1;
+use crate::lgammafn;
+use crate::libc::fabs;
+use crate::libc::DBL_EPSILON;
+use crate::nmath::ml_warn_return_nan;
+use crate::nmath::r_finite;
+use crate::nmath::M_SQRT_2dPI;
+use crate::nmath::DBL_MIN_EXP;
+use crate::nmath::M_LN_SQRT_2PI;
+use crate::pbeta;
+use crate::pnorm;
+use crate::pt;
+use crate::rmath::M_LN2;
+use libm::exp;
+use libm::expm1;
+use libm::fmin;
+use libm::log1p;
+use libm::pow;
+use libm::sqrt;
+
+/// Non-central t distribution
+///
+/// Algorithm AS 243  Lenth,R.V. (1989). Appl. Statist., Vol.38, 185-189.
+///
+/// Cumulative probability at t of the non-central t-distribution
+/// with df degrees of freedom (may be fractional) and non-centrality
+/// parameter delta.
+pub fn pnt(t: f64, df: f64, ncp: f64, lower_tail: bool, log_p: bool) -> f64 {
+    // initialize variables
+    let mut albeta = 0.0;
+    let mut a = 0.0;
+    let mut b = 0.0;
+    let mut del = 0.0;
+    let mut errbd = 0.0;
+    let mut lambda = 0.0;
+    let mut rxb = 0.0;
+    let mut tt = 0.0;
+    let mut x = 0.0;
+    let mut geven = 0.0;
+    let mut godd = 0.0;
+    let mut p = 0.0;
+    let mut q = 0.0;
+    let mut s = 0.0;
+    let mut tnc = 0.0;
+    let mut xeven = 0.0;
+    let mut xodd = 0.0;
+    let mut it = 0;
+    let mut negdel = false;
+    let mut lower_tail = lower_tail;
```

Comment:
you can also set the param in the signature to `mut`

Reply:
Yes 🤦🏻